pkg/broker/deprovision_job.go

```diff
@@ -55,8 +55,26 @@ func (p *DeprovisionJob) Run(token string, msgBuffer chan<- JobMsg) {
 	if err != nil {
 		log.Error("broker::Deprovision error occurred.")
 		log.Errorf("%s", err.Error())
-		msgBuffer <- JobMsg{InstanceUUID: p.serviceInstance.ID.String(), PodName: podName,
-			JobToken: token, SpecID: p.serviceInstance.Spec.ID, Error: err.Error()}
+		// Because we know the error we should return that error.
```

Comment:
I'm not certain I understand why we are returning the specific error message only if `apb.ErrorPodPullErr`. What's the reason for adding this logic?

Reply:
Want to give a specific error about not being able to get the image.

If some other error occurs, I don't want to display that message to the user, that is why I want to do this check here.

Reply:
Thinking more about this. Why don't use just log the error and return a generic error message in `watch_pod` like:

```
log.Errorf("Pod [ %s ] failed - %v", podName, podStatus.Message)
return fmt.Error("Error occurred during APB execution. Please contact administrator if it presists.")
```

Reply:
Because it is not just `watch_pod` that we are worried about, it is the error that apb.(provision/update/deprovision) returns that we are dealing with here.

Reply:
Let's hope that is the biggest oversight I have this year.